projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -324,69 +360,35 @@ export class IgxDateTimeEditorDirective extends IgxMaskDirective implements OnCh
   public onFocus(): void {
     this._isFocused = true;
     this.onTouchCallback();
-    this.updateMask();
+    if (this.value) {
+      this.updateMask();
+    } else {
+      this.inputValue = this.emptyMask;
+    }
     super.onFocus();
   }
 
   /** @hidden @internal */
   public onBlur(value: string): void {
     this._isFocused = false;
-
-    if (this.inputValue === this.emptyMask) {
-      if (this.value) {
-        this.updateValue(null);
-      }
+    if (!this.inputIsComplete() && this.inputValue !== this.emptyMask) {
+      this.updateValue(this.parseDate(this.inputValue));
+    }
+    if (!this.isValidDate(this.value)) {
```

Comment:
after updating `updateMask` method this check will be unnecessary.
on line above, when `inputValue` is not date, `updateValue` will be called with `null` and updated `updateMak` will set the `inputValue` to empty string.